nx-X11/programs/Xserver/hw/nxagent/Args.c

```diff
@@ -187,7 +189,14 @@ int ddxProcessArgument(int argc, char *argv[], int i)
    * Ensure that the options are set to their defaults.
    */
 
-  nxagentProgName = argv[0];
+  char *basec;
+  basec = strdup(argv[0]);
+  nxagentProgName = basename(basec);
```

Comment:
I 'd say use `strdup(basename(basec))` followed by `free(basec)`. This way we have a string that is our own and not something that basename's manpage refers to as "pointers  to  statically allocated memory which may be overwritten by subsequent calls." and  "a pointer to some part of path, so that the string referred to by path should not be modified or freed until the pointer returned by the function is no longer required."

Reply:
This probably also fixes cases where the wrong icon is being displayed, as reported on IRC.